.github/workflows/deploy-production.yml

```diff
@@ -142,13 +150,53 @@ jobs:
       IHE_GW_KEYSTORE_STOREPASS: ${{ secrets.IHE_GW_KEYSTORE_STOREPASS }}
       IHE_GW_KEYSTORE_KEYPASS: ${{ secrets.IHE_GW_KEYSTORE_KEYPASS }}
 
+  ihe-gw-server:
+    if: needs.files-changed.outputs.ihe-gw-server == 'true'
+    needs: files-changed
+    uses: ./.github/workflows/_deploy-ihe-gw.yml
+    with:
+      deploy_env: "production"
+      ECR_REPO_URI: ${{ vars.ECR_REPO_URI_PRODUCTION }}
+      ECS_CLUSTER: ${{ vars.ECS_CLUSTER_PRODUCTION }}
+      IHE_INBOUND_ECS_SERVICE: ${{ vars.IHE_INBOUND_ECS_SERVICE_PRODUCTION }}
+      IHE_OUTBOUND_ECS_SERVICE: ${{ vars.IHE_OUTBOUND_ECS_SERVICE_PRODUCTION }}
+      AWS_REGION: ${{ vars.API_REGION_PRODUCTION }}
+      IHE_GW_ARTIFACT_URL: ${{ vars.IHE_GW_ARTIFACT_URL }}
+      IHE_GW_KEYSTORE_NAME: ${{ vars.IHE_GW_KEYSTORE_NAME }}
+      IHE_GW_ZULUKEY: ${{ vars.IHE_GW_ZULUKEY }}
+      IHE_GW_CONFIG_BUCKET_NAME: ${{ vars.IHE_GW_CONFIG_BUCKET_NAME }}
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      IHE_GW_KEYSTORE_STOREPASS: ${{ secrets.IHE_GW_KEYSTORE_STOREPASS }}
+      IHE_GW_KEYSTORE_KEYPASS: ${{ secrets.IHE_GW_KEYSTORE_KEYPASS }}
+
+  # ihe-gw-config:
+  #   needs: [files-changed, ihe-gw-server]
+  #   if: ${{ !failure() && needs.files-changed.outputs.ihe-gw-config == 'true' }}
+  #   uses: ./.github/workflows/_ihe-gw-push-config.yml
+  #   with:
+  #     deploy_env: "production"
+  #     IHE_GW_CONFIG_BUCKET_NAME: ${{ vars.IHE_GW_CONFIG_BUCKET_NAME }}
+  #     IHE_GW_FULL_BACKUP_LOCATION: ${{ vars.IHE_GW_FULL_BACKUP_LOCATION }}
+  #   secrets:
+  #     IHE_GW_URL: ${{ secrets.IHE_GW_URL_PRODUCTION }}
+  #     IHE_GW_USER: ${{ secrets.IHE_GW_USER }}
+  #     IHE_GW_PASSWORD: ${{ secrets.IHE_GW_PASSWORD }}
+
   release:
     uses: ./.github/workflows/_release.yml
-    needs: [api-sandbox, infra-api-lambdas-sandbox]
+    # TODO 1377 revert this
+    # needs: [api, infra-api-lambdas-sandbox, ihe-gw-config]
+    needs: [api-sandbox, infra-api-lambdas-sandbox, ihe-gw-server]
```

Comment:
why is this necessary?

Reply:
So we only execute release after `ihe-gw-server` is finished - in addition to API and Infra.